lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -584,6 +595,42 @@ def _env_manager_for(self, type: _ServiceType) -> _EnvManager:
         return _EnvManager(file=self._env_file, prefix=type.value)
 
 
+class _JWTKeyManager:
+    """Control the JWT key used by Slurm.
+
+    Todo:
+        Use `jwtctl` to provide backend for generating, setting, and getting
+        JWT key used by `slurmctld` and `slurmrestd`. This way we won't need to
+        pass the `snap` bool as an argument to `__init__`
+    """
+
+    def __init__(self, snap: bool = False):
+        self._keyfile_path = (
+            Path("/var/snap/slurm/common/var/lib/slurm/slurm.state/jwt_hs256.key")
+            if snap
+            else Path("/var/lib/slurm/slurm.state/jwt_hs256.key")
+        )
```

Comment:
This could be a bit less dependent on the snap option by adding a `state_save_location` method to `_OpsManager`, then making our managers return their preferred path for that.

Reply:
Hmm... I think what we really need here is a property such as `var_lib_path` that returns which `/var/lib` path we should be using. Other Slurm configuration options will need to know what the correct `/var/lib` path is - `StateSaveLocation` is shared by all the Slurm daemons - so it might make sense here to expose common paths and let the managers perform the relevant manipulations that they need to.

Reply:
Yep, `var_lib_path` sounds like a nicer solution